storage/src/s3/key_manipulator.cpp

```diff
@@ -0,0 +1,45 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#include "s3/key_manipulator.h"
+#include "string.hpp"
+#include <string>
+
+namespace concord::storage::s3 {
+
+using ::concordUtils::Sliver;
+
+STKeyManipulator::STKeyManipulator(const std::string& prefix) : prefix_(prefix + "/metadata/st/") {}
```

Comment:
Will this work correctly with empty prefix?

If the prefix is empty, it is replaced with a slash (S3 doesn't handle // in urls): https://github.com/vmware/concord-bft/blob/a9e74742d3a77e9b2852d681956f2f965a4697a6/kvbc/include/direct_kv_db_adapter.h#L79

Reply:
fixed, although we always provide a meaningful prefix